drift/views/v0.py

```diff
@@ -1,35 +1,30 @@
-from flask import Blueprint, current_app, jsonify, request
-import os
-import requests
-from requests import RequestException
+from flask import Blueprint, jsonify, request
 
-from drift.views.phony_data import PHONY_DATA
+from drift import info_parser
+from drift.exceptions import InvalidUsage
+from drift.inventory_service_interface import fetch_hosts, get_key_from_headers
 
-# TODO: manage these settings in a config.py
 APP_URL_PREFIX = "/r/insights/platform/drift"
 API_VERSION_PREFIX = "/v0"
-INVENTORY_SVC_URL = os.getenv('INVENTORY_SVC_URL', "http://inventory_svc_url_is_not_set")
-INVENTORY_SVC_PATH = '/r/insights/platform/inventory/api/v1'
 
 section = Blueprint('v0', __name__, url_prefix=APP_URL_PREFIX + API_VERSION_PREFIX)
 
 
 @section.route("/compare")
 def compare():
-    if 'X-RH-IDENTITY' in request.headers:
-        current_app.logger.debug("X-RH-IDENTITY received")
-    return jsonify(PHONY_DATA)
+    host_ids = request.args.getlist('host_ids[]')
+    auth_key = get_key_from_headers(request.headers)
+
+    if host_ids is None or host_ids == []:
+        raise InvalidUsage('host_ids[] not specified', status_code=400)
+
+    if auth_key is None:
+        raise InvalidUsage("auth header not specified", status_code=400)
+
+    comparisons = info_parser.build_comparisons(fetch_hosts(host_ids, auth_key))
+    return jsonify(comparisons)
 
 
 @section.route("/status")
 def status():
-    inventory_connection_status = "fail"
-    try:
-        r = requests.get(INVENTORY_SVC_URL + INVENTORY_SVC_PATH + '/health')
-        if r.status_code == requests.codes.ok:
-            inventory_connection_status = "pass"
-    except RequestException as re:
-        current_app.logger.warn("Unable to contact inventory service: %s" % re)
-        pass
-
-    return jsonify({'status': "running", 'inventory_connection': inventory_connection_status})
+    return jsonify({'status': "running"})
```

Comment:
Dropping the inventory status?

Reply:
ya, I had it there mostly as a baby step to get further on talking to the inventory service. I wasn't sure if I should leave it or not, but I figured we wouldn't want to get an alert on drift service if inventory service was down.

Reply:
Fair. We should think about if inventory service is unreachable how that affects the status of Drift given we rely heavily on inventory.

Reply:
I'll add a card for this. I think as long as we return a 500 and don't retry the connection from drift to inventory it should be OK.

Reply:
(card added, marking as resolved) :card_file_box: